app/lib/katello/util/cdn_var_substitutor.rb

```diff
@@ -90,21 +90,38 @@ def substituable?(path)
         path.include?("$")
       end
 
-      def valid_substitutions?(contnent_url, substitutions)
-        real_path = gsub_vars(contnent_url, substitutions)
+      def valid_substitutions!(content, substitutions)
```

Comment:
I thought `!` was typically reserved for functions that modify the object it's being called upon and not for methods that  could raise an error.